lib/models/Restaurant.js

```diff
@@ -0,0 +1,39 @@
+const pool = require('../utils/pool');
+const { Review } = require('./Review');
+
+class Restaurant {
+  id;
+  name;
+  cuisine;
+  cost;
+  image;
+  website;
+  reviews;
+
+  constructor({ id, name, cuisine, cost, image, website }) {
+    this.id = id;
+    this.name = name;
+    this.cuisine = cuisine;
+    this.cost = cost;
+    this.image = image;
+    this.website = website;
+  }
+
+  static async getAll() {
+    const { rows } = await pool.query('SELECT * from restaurants');
+    return rows.map((row) => new Restaurant(row));
+  }
+
+  static async getById(id) {
+    const { rows } = await pool.query('SELECT * from restaurants WHERE id = $1', [id]);
+    if (!rows) return null;
```

Comment:
`rows` could be an empty array `[]` which is one of the "truthy" values in JS which means you'll never return the `null` value if the getById didn't yield any results. To make sure you're returning null on an empty array you could instead check if `rows` has at least one item in it by accessing the first index.

```js
if (!rows[0]) return null;